core/domain/rights_manager.py

```diff
@@ -199,6 +204,29 @@ def _update_activity_summary(activity_type, activity_rights):
         _update_collection_summary(activity_rights)
 
 
+def update_activity_first_published_msec_if_necessary(
+    activity_type, activity_id, first_published_msec):
+    activity_rights = _get_activity_rights(activity_type, activity_id)
+    if activity_type == ACTIVITY_TYPE_EXPLORATION:
```

Comment:
When I said remove duplication, I meant most of the following lines: you're executing pretty much the same code in both places if I'm not mistaken.